lms/djangoapps/grades/api.py

```diff
@@ -108,7 +108,7 @@ def undo_override_subsection_grade(user_id, course_key_or_id, usage_key_or_id, f
         return
 
     if override is not None and (
-            not feature or feature == override.system):
+            not feature or not override.system or feature == override.system):
         override.delete(feature=feature)
```

Comment:
Should there be an `else` statement here to redirect the code flow to exit this function if no override takes place?  No reason to fire events and update subsection grades.